src/Symfony/Component/Routing/RequestContext.php

```diff
@@ -43,6 +43,12 @@ public function __construct(string $baseUrl = '', string $method = 'GET', string
         $this->setHttpsPort($httpsPort);
         $this->setPathInfo($path);
         $this->setQueryString($queryString);
+
+        if (count($parameters) > 0) {
+            $this->setParameters($parameters);
+        } elseif (count($this->parameters) > 0) {
+            trigger_deprecation('symfony/routing', '7.3', 'Set parameters directly via the own constructor argument');
```

Comment:
do we really need the deprecation?
also, we don't do extra counts when the value already gives the proper result:
```suggestion
        if ($parameters) {
            $this->parameters = $parameters;
        }
```